ToDoApp/ToDoApp/ToDoListViewModel.swift

```diff
@@ -49,25 +49,22 @@ final class ToDoListViewModel: ObservableObject {
     }
     // Handle eddting or creating task when edit or add button is tapped
     private func addEditButtonTapped(mode: CreateTaskViewMode) {
-        let createTaskViewModel: CreateTaskViewModel = CreateTaskViewModel(taskService: taskService, mode: mode) { [weak self] in
+        let createTaskViewModel: CreateTaskViewModel =
+        CreateTaskViewModel(taskService: taskService, mode: mode) { [weak self] in
             self?.resetDestination()
         } onSaved: { [weak self] in
             self?.resetDestination()
             self?.fetchToDoTasks()
         }
         destination = .addTask(createTaskViewModel)
-        
     }
     // Reset destination
     func resetDestination() {
         destination = nil
     }
     // Reload the state when navigate to ToDoListView
     private func fetchToDoTasks() {
-        if case .loading = state {
```

Comment:
Shouldn't we return from the method if the ViewModel is in loading state?

Reply:
I have added if statement to check the condition 
if loading {
return 
}